apollo-api/src/test/java/com/spotify/apollo/ResponseTest.java

```diff
@@ -52,9 +55,39 @@ public void allowsAddingMultipleHeaders() {
         .withHeader("Content-Type", "application/json")
         .withHeaders(headers);
 
-    assertEquals("application/protobuf", response.headers().get("Content-Type"));
-    assertEquals("123", response.headers().get("Content-Length"));
+    assertEquals(Optional.of("application/protobuf"), response.header("Content-Type"));
+    assertEquals(Optional.of("123"), response.header("Content-Length"));
+  }
+
+  @Test
+  public void shouldReturnHeaderCaseInsensitive() throws Exception {
```

Comment:
Yes, I think the test should be implementation agnostic. At least the most important parts.